packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -378,38 +368,43 @@ export class CapabilityService {
     if (!customer || !customer.subscriptions!.data) {
       return [];
     }
-    const subscribedProducts = customer
+    const subscribedPrices = customer
       .subscriptions!.data.filter((sub) =>
         ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
       )
       .flatMap((sub) => sub.items.data)
-      .map(({ price: { product: productId } }) => productId as string);
-    return subscribedProducts;
+      .map(({ price: { id: priceId } }) => priceId as string);
+    return subscribedPrices;
   }
 
   /**
-   * Fetch the list of capabilities for the given product ids.
+   * Fetch the list of capabilities for the given price ids.
    */
-  private async productIdsToClientCapabilities(
-    subscribedProducts: string[]
+  private async priceIdsToClientCapabilities(
+    subscribedPrices: string[]
   ): Promise<ClientIdCapabilityMap> {
     const allCapabilities: Record<string, Set<string>> = {};
-
     // Run through all plans and collect capabilities for subscribed products
-    const plans = await this.stripeHelper.allAbbrevPlans();
-    for (const plan of plans) {
-      if (!subscribedProducts.includes(plan.product_id)) {
+    const prices = await this.stripeHelper.allAbbrevPlans();
+    for (const price of prices) {
+      if (!subscribedPrices.includes(price.plan_id)) {
         continue;
       }
-      const metadata = metadataFromPlan(plan);
```

Comment:
This is the helper that was overwriting any product capabilities I imagine.

Reply:
Exactly, it merged them.

Reply:
Then should we put a fix in `metadataFromPlan` or some other central spot if we should never merge capabilities metadata in that fashion?  That is, after the fix to correctly combine the capabilities metadata, the for loop below will only need to iterate through the plan's metadata.  (Unfortunately we have at least a couple spots where we do the spread operator merge.)

Reply:
I think for most metadata the plans should override the products. Such as translation strings, etc. Capabilities however are the only one I know where we need to extend them.

Reply:
Yes I understand.  I think we leaving the door open to another subtle bug when another engineer doesn't know about these changes and think that `metadataFromPlan` does what its name promises.